tensorflow/python/keras/optimizer_v2/optimizer_v2.py

```diff
@@ -437,6 +437,13 @@ def apply_gradients(self, grads_and_vars, name=None):
         # Distribution strategy does not support reducing an empty list of
         # gradients
         return control_flow_ops.no_op()
+
+      if distribute_ctx.in_cross_replica_context():
+        raise RuntimeError(
+            "`apply_gradients() cannot be called in cross-replica context. "
+            "Use `_distributed_apply()` instead or enter replica context "
```

Comment:
I don't think we should recommend calling a private method. Please remove the "Use `_distributed_apply()` instead" part of the error, and just recommend calling experimental_run_v2.

Reply:
Yes, I agree, I just saw this in the v1 optimizer error message. Removed now. Thanks for the quick feedback!